fix(scap): move machine/agent info to generic platform

gnosek opened the pull request:
/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

/area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

/area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**What this PR does / why we need it**:

The generic platform didn't expose machine_info or agent_info, while we do need this even for source plugins.

Also, allow source plugins to use the live platform (for upcoming syscall plugins).

**Special notes for your reviewer**:

macOS and Windows support is completely untested, fingers crossed for the CI doing its job.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1204#" title="Author self-approved">gnosek</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [gnosek]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

gnosek commented:
cc @incertum

gnosek commented:
also cc @geraldcombs since you seem to be involved with the non-Linux side of things :)

incertum reviewed (dismissed):
Thank you @gnosek 🤩 

/approve

/milestone 0.12.0

poiana commented:
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/falcosecurity/libs/pull/1204#" title="Author self-approved">gnosek</a>*, *<a href="https://github.com/falcosecurity/libs/pull/1204#pullrequestreview-1531777569" title="Approved">incertum</a>*

The full list of commands accepted by this bot can be found [here](https://prow.falco.org/command-help?repo=falcosecurity%2Flibs).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/falcosecurity/libs/blob/master/OWNERS)~~ [gnosek,incertum]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
<!-- META={"approvers":[]} -->

poiana commented:
LGTM label has been added.  <details>Git tree hash: e4a29cd7f6d0b6da8bcf5b9dd1de73fc496ea05e</details>

geraldcombs commented on userspace/libscap/win32/scap_machine_info.c:
```diff
@@ -0,0 +1,113 @@
+	{
+		OSVERSIONINFOA os_info;
+		os_info.dwOSVersionInfoSize = sizeof(OSVERSIONINFO);
+		if(GetVersionExA(&os_info))
```
Note that this [probably won't return the correct version](https://learn.microsoft.com/en-us/windows/win32/api/sysinfoapi/nf-sysinfoapi-getversionexa?redirectedfrom=MSDN) on modern versions of Windows. We use RtlGetVersion in Wireshark, but it looks like you can also call GetFileVersionInfo on one of the kernel DLLs.

gnosek replied:
I'm happy it compiled without me seeing a Windows machine ;) I'm even happier to replace it with something that actually works.

For future me: https://github.com/wireshark/wireshark/blob/dcc9cbffefe0598eadec11cbc1869943a26219c5/wsutil/os_version_info.c#L192

(I don't think we need anything *that* fancy but I'll use it for inspiration ;])

geraldcombs replied:
@incertum I think this falls in the "future PR" category.

incertum replied:
@gnosek I'll re-approve tomorrow as we need a rebase now!

FedeDP commented:
/milestone 0.12.0

poiana commented:
New changes are detected. LGTM label has been removed.

gnosek commented:
@incertum, rebased.

@geraldcombs, I did a yolo implementation using RtlGetVersion, please take a look if you can :) (taken from Wireshark, simplified a lot with less detail)

FedeDP reviewed (approved):
/approve

poiana commented:
LGTM label has been added.  <details>Git tree hash: 76b7e2955e38508dbd2c75ce9865bfd7ca7e1747</details>

incertum reviewed (approved):
/approve
